src/utils/analytics.ts

```diff
@@ -31,15 +31,15 @@ export function getUtm() {
 export const setupAnalytics = () => {
   setUtm()
 }
-
-export const trackEvent = async (type: string, chain: string) => {
+export const trackEvent = async (type: string, chain: string, customProperties?: any) => {
   const referrer = getUtm()
+
+  const props = referrer ? { ...customProperties, referrer } : customProperties
```

Comment:
I don't think we need *referrer ? {} : {}*, we can just pass { ...customProperties, referrer}